format.js

```diff
@@ -158,20 +217,45 @@ function makeEncryptionFormat() {
     return ciphertextBuf
   }
 
+  function selfDecryptionKeys(authorId) {
+    const selfKeys = keyring.self.get()
+    if (keyring.signing.has(authorId)) return [selfKeys]
+    else if (legacyMode && authorId === mainKeys.id) return [selfKeys]
+    else return []
+  }
+
+  function dmDecryptionKeys(authorId) {
+    if (legacyMode) {
+      const dmKeys = keyring.dm.get(mainKeys.id, authorId)
+      if (!dmKeys) addDMPairSync(mainKeys, authorId)
+      if (!keyring.dm.has(mainKeys.id, authorId)) return []
+      return [keyring.dm.get(mainKeys.id, authorId)]
+    } else {
+      const myRootKeys = keyring.signing.get('root')
+      if (!myRootKeys) return []
+      const myLeafId = keyring.dm.triangulate(myRootKeys.id, authorId)
+      if (!myLeafId) return []
+      if (!keyring.dm.has(myLeafId, authorId)) return []
+      return [keyring.dm.get(myLeafId, authorId)]
+    }
+  }
+
   function decrypt(ciphertextBuf, opts) {
     const authorId = opts.author
     const authorBFE = BFE.encode(authorId)
     const previousBFE = BFE.encode(opts.previous)
 
-    const { self, dm, group, poBox } = _keyring.decryptionKeys(authorId)
+    const group = keyring.group.list().map(keyring.group.get)
```

Comment:
I know this is just matching the old code, but could we call this groups instead?

Reply:
I called them now `groupKeys`, `selfKey`, `dmKey`